### ISSUE TITLE
Animations should disable or degrade when battery saver mode is enabled (Android)

### ISSUE THREAD

**Issue by lukef** [flutter/flutter#11436]
Native Android apps/libraries will disable certain/most animation when the user enters low-battery/battery saver mode to preserve battery life. Flutter should do so as well.

For example, enable battery saver mode and try navigating around in Google Newsstand. Progress spinners degrade, page transitions are disabled, etc.

**Comment by eseidelGoogle** [comment]
Agreed, we should support this.  It probably will be a while until we do.  This is also needed for Accessibility.  @goderbauer

**Comment by jonahwilliams** [comment]
For the native material widgets I have seen it seems that the widget is responsible for determining whether or not it should animate for accessibility:

https://github.com/material-components/material-components-android/blob/7aff9d2e8b5d62d3d7c554d3debfe9f1a11a7c77/lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java#L495

We might need to do something similar for battery saver mode, possibly by exposing some sort of `shouldAnimate` getter on a binding class and letting our components decide what to do.  We could also configure our animations to look at this member and skip to the end, but it might not be trivial to do so in such a way that applications don't break.  Especially animations that repeat using imperative code.

**Comment by jonahwilliams** [comment]
actually, just changing a setting in `AnimationController` is working surprisingly well.  I will investigate the cases where assertions are thrown in the gallery app to cover the edge cases.  routing seems to be a big issue

**Comment by creativecreatorormaybenot** [comment]
@jonahwilliams I just did some testing on `AnimationBehavior.normal` (and `SemanticsBinding.instance!.disableAnimations` in that regard) and I found out that the behavior does not match this issue on Android.

What I expected:
* Animations would not run when battery saver is enabled → but looking into the code, I found out it would only speed it up _anyway_.
* Then, animations to speed up when battery saver is enabled → does not happen **at all**.
* Okay, then at least `SemanticsBinding.instance!.disableAnimations` to return `true` when battery saver is enabled → nope.
* Using `Accessibility settings → Remove animations` in the settings should make `SemanticsBinding.instance!.disableAnimations` return `true` → yes, happens.

So overall: why did you close this? The PR (#20354) does not affect battery saver at all (tested on Android).
